From 7dd57a7f78a4effd417e0100cb3a7153d5534766 Mon Sep 17 00:00:00 2001
From: topjohnwu <topjohnwu@gmail.com>
Date: Sun, 14 Jul 2024 16:55:01 -0700
Subject: [PATCH] Add missing sigisemptyset implementation


diff --git a/shell/hush.c b/shell/hush.c
index 226a6d68f..0ab3e68dc 100644
--- a/shell/hush.c
+++ b/shell/hush.c
@@ -2081,6 +2081,15 @@ enum {
 		,
 };
 
+#ifdef __BIONIC__
+static int sigisemptyset(const sigset_t *set)
+{
+    sigset_t empty;
+    sigemptyset(&empty);
+    return memcmp(&empty, set, sizeof(sigset_t)) == 0;
+}
+#endif
+
 static void record_pending_signo(int sig)
 {
 	sigaddset(&G.pending_set, sig);
-- 
2.50.1

